app.js

```diff
@@ -1389,6 +1401,49 @@ const getPossibleTimestampsRecursive = async (obj) =>{
   return arr;
 }
 
+const parseLocation = (data) => {
+  const latitudeStr = data?.['Profile-EXIF']?.['GPS Latitude'];
+  const longitudeStr = data?.['Profile-EXIF']?.['GPS Longitude'];
+  const longRef = data?.['Profile-EXIF']?.["GPS Longitude Ref"];
+  const latRef = data?.['Profile-EXIF']?.["GPS Latitude Ref"];
+
+  // Missing exif data
+  if (!latitudeStr || !longitudeStr) {
+    return {
+      latitude: undefined,
+      longitude: undefined
+    }
+  }
+
+  // GPS coordinates are formatted as rationals for degrees, minutes, and seconds
+  // E.g. GPS Latitude: '33/1,23/1,4831/100'
+  function stringToDecimal(gpsString) {
```

Comment:
i think it would be best to wrap some of this a `try... catch` or similar, just to handle the case where unexpected exif data is parsed from the image. for example, catching where `numerator` and/or `denominator` are undefined or cause divide-by-zero error. or when `parts.length < 3` etc.

Reply:
Very good point, this code makes a lot of assumptions about consistent formatting. Updated with improved error handling.